Solutions

SergiyVS opened the pull request.

SergiyVS committed:
Solutions

Anastasiia-Svintsova commented on src/transformState.js:
```diff
@@ -5,7 +5,29 @@
+        break;
+
+      case 'addProperties':
+        for (const extraKey in action.extraData) {
+          state[extraKey] = action.extraData[extraKey];
+        }
```
Use [Object.assign()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) instead

Anastasiia-Svintsova commented on src/transformState.js:
```diff
@@ -5,7 +5,29 @@
+        }
+        break;
+
+      default: ;
```
Add some action for a default case 
![image](https://github.com/mate-academy/js_stateful-object/assets/99403198/d0624c60-bade-4562-a315-3ae2b36ac13d)

Anastasiia-Svintsova commented on src/transformState.js:
```diff
@@ -5,7 +5,29 @@
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
```
Use destructuring for the action object

Anastasiia-Svintsova commented on src/transformState.js:
```diff
@@ -5,7 +5,29 @@
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
```
It's a good practice to move string values to variables to avoid errors connected with typos and to promote code maintainability and readability.

Anastasiia-Svintsova reviewed (changes requested):
Good start! Check my comments above

SergiyVS committed:
1. for...in loop changed to Object.assign()
2.  Added message about wrong action
3. Prop of action object assigned to variable
4. Strings in switch-case changed to variables

agniya-i commented on src/transformState.js:
```diff
@@ -5,7 +5,35 @@
 function transformState(state, actions) {
-  // write code here
+  const actionClear = 'clear';
+  const actionRemProp = 'removeProperties';
```
```suggestion
  const actionRemoveProp = 'removeProperties';
```

SergiyVS committed:
const actionRemProp changed to const actionRemoveProp
